src/lib.rs

```diff
@@ -17,87 +21,189 @@ impl Default for Syntax {
     }
 }
 
+/// A field frequency
 #[derive(Debug, Clone)]
 pub enum Frequency {
+    /// A well-formed message can have zero or one of this field (but not more than one).
     Optional,
+    /// This field can be repeated any number of times (including zero) in a well-formed message.
+    /// The order of the repeated values will be preserved.
     Repeated,
+    /// A well-formed message must have exactly one of this field.
     Required,
 }
 
+/// Protobuf supported field types
 #[derive(Debug, Clone, PartialEq, Eq, Hash)]
 pub enum FieldType {
+    /// Protobuf int32
+    ///
+    /// # Remarks
+    ///
+    /// Uses variable-length encoding. Inefficient for encoding negative numbers – if
+    /// your field is likely to have negative values, use sint32 instead.
     Int32,
+    /// Protobuf int64
+    ///
+    /// # Remarks
+    ///
+    /// Uses variable-length encoding. Inefficient for encoding negative numbers – if
+    /// your field is likely to have negative values, use sint64 instead.
     Int64,
+    /// Protobuf uint32
+    ///
+    /// # Remarks
+    ///
+    /// Uses variable-length encoding.
     Uint32,
+    /// Protobuf uint64
+    ///
+    /// # Remarks
+    ///
+    /// Uses variable-length encoding.
     Uint64,
+    /// Protobuf sint32
+    ///
+    /// # Remarks
+    ///
+    /// Uses variable-length encoding. Signed int value. These more efficiently
+    /// encode negative numbers than regular int32s.
     Sint32,
+    /// Protobuf sint64
+    ///
+    /// # Remarks
+    ///
+    /// Uses variable-length encoding. Signed int value. These more efficiently
+    /// encode negative numbers than regular int32s.
     Sint64,
+    /// Protobuf bool
     Bool,
+    /// Protobuf enum (holds the enum name)
     Enum(String),
+    /// Protobuf fixed64
+    ///
+    /// # Remarks
+    ///
+    /// Always eight bytes. More efficient than uint64 if values are often greater than 2^56.
     Fixed64,
+    /// Protobuf sfixed64
+    ///
+    /// # Remarks
+    ///
+    /// Always eight bytes.
     Sfixed64,
+    /// Protobuf double
     Double,
+    /// Protobuf string
+    ///
+    /// # Remarks
+    ///
+    /// A string must always contain UTF-8 encoded or 7-bit ASCII text.
     String_,
```

Comment:
Probably this field could be named `String`, not `String_`, because enum variants and type names live in different namespaces.